microservices/site-launch/lambda-functions/primary-domain-validation/index.ts

```diff
@@ -0,0 +1,89 @@
+import dns from "dns"
+
+import {
+  AmplifyClient,
+  GetDomainAssociationCommand,
+  DomainStatus,
+} from "@aws-sdk/client-amplify"
+import type {
+  GetDomainAssociationCommandInput,
+  GetDomainAssociationCommandOutput,
+} from "@aws-sdk/client-amplify"
+
+import logger from "../../shared/logger"
+import {
+  SITE_LAUNCH_LAMBDA_TYPE,
+  SITE_LAUNCH_LAMBDA_STATUS,
+} from "../../shared/types" // TODO: Change to aliased imports
+
+interface PrimaryDomainValidationLambdaParams {
+  appId: string
+  primaryDomain: string
+  cloudfrontDomain: string
+}
+
+interface PrimaryDomainValidationLambdaResponse {
+  lambdaType: SITE_LAUNCH_LAMBDA_TYPE
+  status: SITE_LAUNCH_LAMBDA_STATUS
+  appId: string
+  primaryDomain: string
+}
+
+export const primaryDomainValidation = async (
+  event: PrimaryDomainValidationLambdaParams
+): Promise<PrimaryDomainValidationLambdaResponse> => {
+  logger.info(event)
+
+  const { AWS_REGION } = process.env
+  const amplifyClient = new AmplifyClient({
+    region: AWS_REGION,
+  })
+
+  // Validation check
+  const { appId, primaryDomain, cloudfrontDomain } = event
+
+  const params: GetDomainAssociationCommandInput = {
+    appId,
+    domainName: primaryDomain,
+  }
+  const getDomainAssociationCommand = new GetDomainAssociationCommand(params)
+
+  try {
```

Comment:
wdyt about just inlining the `logger.error(error)` call and removing the `try/catch` block?

Reply:
we could but we would have to throw them regardless to trigger the failure lambda! I would prefer this over throwing and logging each error, wdyt?

Reply:
hmm after some research, it seems like if we log on the failure notif lambda, this code would be clearer as we don't have to `catch` the error we throw and log it. (error handling code is separate from business logic)

if you opt to omit the refactor or do it down the line, i think this is fine

Reply:
f85fdb6